app/models/product.rb

```diff
@@ -77,7 +77,7 @@ def self.types
   end
 
   def self.mergeable_types
-    @mergeable_types ||= default_types
+    default_types
```

Comment:
Depending on what order we put things in, I think we'll have some merge conflicts here.